src/Distribution/Nixpkgs/Haskell/FromCabal.hs

```diff
@@ -1,3 +1,4 @@
+{-# LANGUAGE CPP #-}
```

Comment:
I really don't want to support multiple versions of Cabal, especially if that support requires CPP. I am sorry, but `cabal2nix` is inherently tied to Nixpkgs and it's very easy to build this tool within Nixpkgs. Adding complexity to the build to enable additional methods to build this tool is just not a priority for me.

Reply:
@peti I understand that was your original motivation, but I firmly believe that being compatible with the latest LTS will get you many more contributors (such as Robert from #431, me, and so on), and that the time you will save by having people help out will dwarf the additional effort (e.g. between Cabal 3.0 and 3.2, it is only 2 lines).

Also, the differences would likely be quite small, as the _latest_ LTS is enough (we don't need to keep compat with older ones).

I propose to roll with it for now, we can still throw it away again if we do find that the maintenance burden is large.

Reply:
OK.